build-logic/binary-compatibility/src/main/groovy/gradlebuild/binarycompatibility/rules/UpgradePropertiesRuleSetup.java

```diff
@@ -51,28 +52,29 @@ public void execute(ViolationCheckContext context) {
         List<UpgradedProperty> currentUpgradedProperties = UpgradedProperties.parse(params.get(CURRENT_UPGRADED_PROPERTIES_KEY));
         List<UpgradedProperty> baselineUpgradedProperties = UpgradedProperties.parse(params.get(BASELINE_UPGRADED_PROPERTIES_KEY));
         context.putUserData(CURRENT_ACCESSORS_OF_UPGRADED_PROPERTIES, diff(
-            mapCurrentAccessorsOfUpgradedProperties(currentUpgradedProperties),
-            mapCurrentAccessorsOfUpgradedProperties(baselineUpgradedProperties)
+            currentAccessorsOfUpgradedProperties(currentUpgradedProperties),
+            currentAccessorsOfUpgradedProperties(baselineUpgradedProperties)
         ));
-        context.putUserData(OLD_ACCESSORS_OF_UPGRADED_PROPERTIES, diff(
-            mapOldAccessorsOfUpgradedProperties(currentUpgradedProperties),
-            mapOldAccessorsOfUpgradedProperties(baselineUpgradedProperties)
+        context.putUserData(OLD_REMOVED_ACCESSORS_OF_UPGRADED_PROPERTIES, diff(
+            oldRemovedAccessorsOfUpgradedProperties(currentUpgradedProperties),
+            oldRemovedAccessorsOfUpgradedProperties(baselineUpgradedProperties)
         ));
-        context.putUserData(SEEN_OLD_ACCESSORS_OF_UPGRADED_PROPERTIES, new HashSet<>());
+        context.putUserData(SEEN_OLD_REMOVED_ACCESSORS_OF_UPGRADED_PROPERTIES, new HashSet<>());
     }
 
-    private static Map<AccessorKey, UpgradedProperty> mapCurrentAccessorsOfUpgradedProperties(List<UpgradedProperty> upgradedProperties) {
+    private static Map<AccessorKey, UpgradedProperty> currentAccessorsOfUpgradedProperties(List<UpgradedProperty> upgradedProperties) {
         return upgradedProperties.stream().collect(Collectors.toMap(AccessorKey::ofUpgradedProperty, Function.identity()));
     }
 
-    private static Map<AccessorKey, ReplacedAccessor> mapOldAccessorsOfUpgradedProperties(List<UpgradedProperty> upgradedProperties) {
+    private static Map<AccessorKey, ReplacedAccessor> oldRemovedAccessorsOfUpgradedProperties(List<UpgradedProperty> upgradedProperties) {
         return upgradedProperties.stream()
-            .flatMap(UpgradePropertiesRuleSetup::mapOldAccessorsOfUpgradedProperty)
+            .flatMap(UpgradePropertiesRuleSetup::oldRemovedAccessorsOfUpgradedProperty)
             .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
     }
 
-    private static Stream<Map.Entry<AccessorKey, ReplacedAccessor>> mapOldAccessorsOfUpgradedProperty(UpgradedProperty upgradedProperty) {
+    private static Stream<Map.Entry<AccessorKey, ReplacedAccessor>> oldRemovedAccessorsOfUpgradedProperty(UpgradedProperty upgradedProperty) {
         return upgradedProperty.getReplacedAccessors().stream()
+            .filter(replacedAccessor -> replacedAccessor.getBinaryCompatibility() == ACCESSORS_REMOVED)
```

Comment:
💭 This is the main change: when we tried to figure out what methods were already removed in baseline and what were removed current version, we didn't take in to account that some methods could also be "upgraded" but "KEPT", like it's the case with `@ToBeReplacedByLazyProperty` instrumentation